models/magiceden/solana/magiceden_solana_trades.sql

```diff
@@ -0,0 +1,243 @@
+            , tk.token_name
+            , tk.token_symbol
+            , 'secondary' as trade_type
+            , token_size as number_of_items --all single trades right now
+            , t.trade_category
+            , t.account_buyer as buyer
+            , t.account_seller as seller
+            , t.price as amount_raw --magiceden does not include fees in the emitted price
+            , t.price/1e9 as amount_original
+            , t.price/1e9 * sol_p.price as amount_usd
+            , 'SOL' as currency_symbol
+            , 'So11111111111111111111111111111111111111112' as currency_address
+            , t.account_metadata --token id equivalent
+            , tk.account_master_edition --token id equivalent
+            , tk.account_mint --token id equivalent
+            , tk.verified_creator --contract address equivalent
+            , tk.collection_mint --contract address equivalent
+            , 'TSWAPaqyCSx2KABk68Shruf4rp7CxcNi8hAsbdwmHbN' as project_program_id
+            , cast(null as varchar) as aggregator_name
+            , cast(null as varchar) as aggregator_address
+            , t.call_tx_id as tx_id
+            , t.call_block_slot as block_slot
+            , t.call_tx_signer as tx_signer
+            , t.taker_fee as taker_fee_amount_raw --taker fees = platform fees
+            , t.taker_fee/1e9 as taker_fee_amount
+            , t.taker_fee/1e9 * sol_p.price as taker_fee_amount_usd
+            , case when t.taker_fee = 0 then 0 else t.taker_fee/t.price end as taker_fee_percentage
+            , t.maker_fee as maker_fee_amount_raw
+            , t.maker_fee/1e9 as maker_fee_amount
+            , t.maker_fee/1e9 * sol_p.price as maker_fee_amount_usd
+            , case when t.maker_fee = 0 then 0 else t.maker_fee/t.price end as maker_fee_percentage
+            , cast(null as double) as amm_fee_amount_raw
+            , cast(null as double) as amm_fee_amount
+            , cast(null as double) as amm_fee_amount_usd
+            , cast(null as double) as amm_fee_percentage
+            , t.royalty as royalty_fee_amount_raw 
+            , t.royalty/1e9 as royalty_fee_amount
+            , t.royalty/1e9 * sol_p.price as royalty_fee_amount_usd
+            , case when t.royalty = 0 then 0 else t.royalty/t.price end as royalty_fee_percentage
+            , t.instruction
+            , t.outer_instruction_index
+            , coalesce(t.inner_instruction_index,0) as inner_instruction_index
+        FROM trades t
+        LEFT JOIN {{ ref('tokens_solana_nft') }} tk
+            ON t.account_metadata = tk.account_metadata
+        LEFT JOIN {{ source('prices', 'usd') }} sol_p ON sol_p.blockchain = 'solana' and sol_p.symbol = 'SOL' and sol_p.minute = date_trunc('minute', t.call_block_time) --get sol_price
+    )
+
+SELECT 
+*
+, concat(project,'-',trade_category,'-',cast(outer_instruction_index as varchar),'-',cast(coalesce(inner_instruction_index, 0) as varchar),'-',account_metadata,'-',tx_id) as unique_trade_id
```

Comment:
why is this column needed and/or wanted?

Reply:
ah it was from testing, I'll remove